File: example/GetSocialNotificationServiceExtension/NotificationService.m
//
//  NotificationService.m
//  NotificationExtension
//
//  Created by Gábor Vass on 19/03/2021.
//

#import "NotificationService.h"
#if !TARGET_IPHONE_SIMULATOR
#import <GetSocialNotificationExtension/GetSocialNotificationExtension.h>
#endif
@interface NotificationService ()

#if !TARGET_IPHONE_SIMULATOR
@property (nonatomic, strong) GetSocialNotificationRequestHandler *handler;
#endif
@end

@implementation NotificationService

- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
#if !TARGET_IPHONE_SIMULATOR
	self.handler = [GetSocialNotificationRequestHandler new];
	[self.handler handleNotificationRequest:request withContentHandler:contentHandler];
#endif
}

- (void)serviceExtensionTimeWillExpire {
#if !TARGET_IPHONE_SIMULATOR
	[self.handler serviceExtensionTimeWillExpire];
#endif
}

@end

File: example/GetSocialNotificationServiceExtension/NotificationService.swift
//
//  NotificationService.swift
//  ServiceExtension
//
//  Copyright © 2020 GetSocial. All rights reserved.
//

import UserNotifications

class NotificationService: UNNotificationServiceExtension {

    var handler: GetSocialNotificationExtensionHandler?

    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
        self.handler = GetSocialNotificationExtensionHandler.init()
        self.handler?.handle(request, withContentHandler: contentHandler)
    }
    
    override func serviceExtensionTimeWillExpire() {
        self.handler?.serviceExtensionTimeWillExpire()
    }

}
